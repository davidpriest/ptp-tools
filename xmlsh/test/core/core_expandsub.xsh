# core_expandsub.xsh
# test expanding of subprocess aka $()


a=$(echo foo)
echo Should be foo: $a

a=$(echo a; echo b ; echo c;)
echo Should be 3 : ${#a}

b=$(echo a1 ; echo b1 ; echo c1 ; )
for c in $b ; do
	echo Arg $c
done

# Test return values in $()
A=$(true)  || echo FAIL true should be success
A=$(false) || echo Success expected false
A=$(true;false)
echo retval is $?
unset A

# Test backticks
A=`echo foo1`
[ "$A" = "foo1" ] || echo FAIL should be foo1
 
