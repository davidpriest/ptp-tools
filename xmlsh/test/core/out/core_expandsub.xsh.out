Should be foo: foo
Should be 3 : 3
Arg a1
Arg b1
Arg c1
Success expected false
retval is 1
